DigitalLearningSolutions.Data/Helpers/ExternalApis/LearningHubApiClient.cs

```diff
@@ -0,0 +1,153 @@
+    {
+        Task<ResourceSearchResult> SearchResource(
+            string text,
+            int? offset = null,
+            int? limit = null,
+            IEnumerable<string>? resourceTypes = null
+        );
+
+        Task<ResourceReferenceWithReferenceDetails> GetResourceByReferenceId(int resourceReferenceId);
+
+        Task<BulkResourceReferences> GetBulkResourcesByReferenceIds(
+            IEnumerable<int>? resourceReferenceIds = null
+        );
+    }
+
+    public class LearningHubApiClient : ILearningHubApiClient
+    {
+        private readonly HttpClient client;
+        private readonly string learningHubApiBaseUrl;
+
+        public LearningHubApiClient(HttpClient httpClient, IConfiguration config)
+        {
+            string learningHubApiKey = config.GetLearningHubApiKey();
+            learningHubApiBaseUrl =
+                "https://uks-learninghubnhsuk-openapi-dev.azurewebsites.net";
+
+            client = httpClient;
+            client.DefaultRequestHeaders.Accept.Clear();
+            client.DefaultRequestHeaders.Add("X-API-KEY", learningHubApiKey);
+        }
+
+        public async Task<ResourceSearchResult> SearchResource(
+            string text,
+            int? offset = null,
+            int? limit = null,
+            IEnumerable<string>? resourceTypes = null
+        )
+        {
+            const string searchEndpoint = "/Resource/Search";
+
+            var endpointPath = learningHubApiBaseUrl + searchEndpoint;
+            var requestUrl = GetUrlWithSearchQueryParams(endpointPath, text, offset, limit, resourceTypes);
+
+            var response = await client.GetStringAsync(requestUrl);
+            var result = JsonConvert.DeserializeObject<ResourceSearchResult>(response);
+            return result;
+        }
+
+        public async Task<ResourceReferenceWithReferenceDetails> GetResourceByReferenceId(int resourceReferenceId)
+        {
+            const string resourceEndpoint = "/Resource/";
+            var requestUrl = learningHubApiBaseUrl + resourceEndpoint + resourceReferenceId;
+
+            var response = await client.GetStringAsync(requestUrl);
+            var result = JsonConvert.DeserializeObject<ResourceReferenceWithReferenceDetails>(response);
+            return result;
+        }
+
+        public async Task<BulkResourceReferences> GetBulkResourcesByReferenceIds(
+            IEnumerable<int>? resourceReferenceIds = null
+        )
+        {
+            const string bulkEndpoint = "/Resource/Bulk";
+            var endpointPath = learningHubApiBaseUrl + bulkEndpoint;
+            var requestUrl = GetUrlWithResourceReferenceIdQueryParams(endpointPath, resourceReferenceIds);
+
+            var response = await client.GetStringAsync(requestUrl);
+            var result = JsonConvert.DeserializeObject<BulkResourceReferences>(response);
+            return result;
+        }
+
+        private static string GetUrlWithSearchQueryParams(

```

Comment:
I'm pretty sure there must be a better way to do all this query string construction without using a bunch of string concatenation and interpolation.

Reply:
I initially used QueryHelpers.AddQueryString to add the query strings, but this method only accepts string values, not StringValues values, so it was impossible to add the query strings dealing with enumerables. Steve suggested I create the query strings manually, but if you can think of a better way, please enlighten me. It does feel a bit icky